qadence/composer/composer.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+from pydantic import BaseModel, validator
+
+from qadence.types import BackendName, DiffMode
+
+
+class ExperimentComposer(BaseModel):
+    BACKEND: BackendName
+    DIFF_MODE: DiffMode
+
+    @validator("BACKEND")
+    @classmethod
+    def validate_backend(cls, value: BackendName) -> BackendName:
+        return BackendName(value)
+
+    @validator("DIFF_MODE")
+    @classmethod
+    def validate_diffmode(cls, value: DiffMode, values: dict) -> DiffMode:
+        validated_diffmode = DiffMode(value)
+        if (
+            validated_diffmode == DiffMode.AD
+            and (backend := values["BACKEND"]) != BackendName.PYQTORCH
+        ):
+            raise ValueError(f"Backend {backend} does not support diff_mode {validated_diffmode}.")
+        return validated_diffmode
```

Comment:
Also (and this is likely for @dominikandreasseitz as well, do we want to check the configuration as well here? I am thinking about setting parameters such as `_use_gate_params` in here to make sure it all checks out. If we write the validation logic into a separate routine, we can reuse it in the `backend_factory` to avoid any code duplication.

Reply:
That's a side work that would fit here I think. Essentially setting the backend config but there might be others I don't immediately see now.

Reply:
Done.